### PR TITLE
Run unit tests across Linux, OSX and Windows using Jenkins

### PR DESCRIPTION
**Pull request by ewanharris**
Part of https://jira.appcelerator.org/browse/DAEMON-202

This uses the Jenkins setup to run the unit tests across Linux, OSX and Windows, it only runs the tests. It does not publish the package for us, we could add that later if we wish. 

Additionally, this same setup will more than likely be used across androidlib, ioslib, jdklib and windowslib, so it would make make sense to migrate this into a common utils (I think the pipeline-library)

@cb1kenobi I changed two bits in the tests to get this passing,

- Made sure that the mocks/empty dir exists as it sometimes went missing when un-stashing sources on OSX and Windows
- Removed the bash builtin from the 1.8-32-bit javac mock

### CODE REVIEW

File: test/test-jdk.js [hunk 1]
```diff
@@ -25,7 +26,11 @@ describe('JDK', () => {
 	});
 
 	it('should error if dir is missing the JVM library', () => {
-		return detect(path.join(__dirname, 'mocks', 'empty'))
+		const mockDir = path.join(__dirname, 'mocks', 'empty');
+		if (!fs.existsSync(mockDir)) {
+			fs.mkdirSync(mockDir);
```
Comment (cb1kenobi) [inline test/test-jdk.js]: I don't understand. The `empty` directory does exist: https://github.com/appcelerator/jdklib/tree/master/test/mocks/empty.

Comment (ewanharris) [reply test/test-jdk.js]: Yeah, it does in the repo, it does when we unstash on linux, but when we unstash on OSX/Windows it doesn't, despite using `allowEmpty: true` in the stash (maybe I'm using that wrong, I dunno), see the failures in [this run](https://jenkins.appcelerator.org/blue/organizations/jenkins/cli%2Fjdklib/detail/jenkinsfile/9/pipeline/22) for example

I guess I should add a comment clarifying why I added that.

Comment (ewanharris) [reply test/test-jdk.js]: So I think I was using it wrong. I can remove this :)

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+           fingerprint 'package.json'
+            try {
+              if('windows'.equals(os)) {
+                bat 'yarn run coverage'
+              } else {
+                sh 'yarn run coverage'
+              }
+            } finally {
+              // record results even if tests/coverage 'fails'
+              junit 'junit.xml'
```
Comment (sgtcoolguy) [inline Jenkinsfile]: You may want to "hack" the generated test report to differentiate the tests for each OS. It won't be easy to determine which OS failed a given test when they're all merged with the same names. For the SDK I inject the platform name as a prefix to the test names.

Not sure if there's a better/more official way of handling running the same tests on multiple platforms and merging them together via JUnit/Jenkins.

**Pull request merged**